whois-update/src/main/java/net/ripe/db/whois/update/domain/UpdateMessages.java

```diff
@@ -656,6 +656,19 @@ public static Message multipleUserMntBy(final Collection<CIString> userMntners)
         return new Message(Type.ERROR, "Multiple user-'mnt-by:' are not allowed, found are: '%s'", Joiner.on(", ").join(userMntners));
     }
 
+    public static Message notValidSource() {
+        return new Message(Type.ERROR, "Can not set NONAUTH source when aut-num is not NONAUTH");
+    }
```

Comment:
If you add an error instead of changing something, it should be in validator not transfer. Also check SingleUpdateHandler, we look for update error message only in businessValidator rule, transfer should not throw validation error , it should just only transform object.

Reply:
ok, I will move this

Reply:
removed this, as we talk in by zoom